### ISSUE TITLE
When i just run train.py,something was wrong

### ISSUE THREAD

**Issue by Zengpeizhi** [whywhs/Pytorch-Handwritten-Mathematical-Expression-Recognition#2]
![CAE$OHL(1OO KSRO %0 `6U](https://user-images.githubusercontent.com/38500882/54663587-bdfaab80-4b1c-11e9-807c-b14f3090073e.png)
@whywhs  hello,could you please to help me  to solve the problem?(i just follow you readme and when i run Train.py ,it was wrong)

**Comment by whywhs** [comment]
Of course. I just ran my program on computer but there was no error, so I  am not sure what caused your problems. Could you please show me the  complete error message? Or just show me where the error occurred in my program.

**Comment by Zengpeizhi** [comment]
Thank you very much for your reply! The code error information is shown in the attachment.I just follow your instruction and run gen_pkl.py,and it work.But after I run Train.py,the mistake happen.




------------------ 原始邮件 ------------------
发件人: "Hongyu Wang"<notifications@github.com>;
发送时间: 2019年3月20日(星期三) 下午3:03
收件人: "whywhs/Pytorch-Handwritten-Mathematical-Expression-Recognition"<Pytorch-Handwritten-Mathematical-Expression-Recognition@noreply.github.com>;
抄送: "曾培智"<1069319173@qq.com>; "Author"<author@noreply.github.com>; 
主题: Re: [whywhs/Pytorch-Handwritten-Mathematical-Expression-Recognition]When i just run train.py,something was wrong (#2)




Of course. I just ran my program on computer but there was no error, so I  am not sure what caused your problems. Could you please show me the  complete error message? Or just show me where the error occurred in my program.

**Comment by whywhs** [comment]
I see your pictures, but I can't see where the error occurred in my program. So, maybe you can take a complete screenshot of the error. Is your pytorch version 0.3?

**Comment by Zengpeizhi** [comment]
> I see your pictures, but I can't see where the error occurred in my program. So, maybe you can take a complete screenshot of the error. Is your pytorch version 0.3?

oh，sorry,my pytorch is 1.0.1 .So the error was because of  that?If so,I try to use 0.3

**Comment by whywhs** [comment]
yes! I've been finished this project about 1 year ago, so the pytorch version is 0.3

**Comment by Zengpeizhi** [comment]
> yes! I've been finished this project about 1 year ago, so the pytorch version is 0.3

@whywhs  Thank you very much!When I use pytorch0.3,it runs!

**Comment by whywhs** [comment]
Furthermore, you can write num_works in Train.py (line 128, line 135) if you want because this can make your training faster than before.

**Comment by Zengpeizhi** [comment]
hello，thank you very much for your help! And if i want to improve the result ,what do you think i should  do?Can you give me some advice?

**Comment by whywhs** [comment]
Sure. I think you can get a better result if you can do these:
1. When you finished your training, you can fix the weight of the encoder and train the decoder separately, this will improve your result about 3-4%.
2. If you want a better result, the batch_size is important. Meanwhile, you can also get a faster speed.
3. The teacher-forcing ratio is also important, you can try some higher ratio if you are interested.
